addons/project/populate/project.py

```diff
@@ -57,15 +57,37 @@ def get_stage_ids(random, **kwargs):
 class ProjectTask(models.Model):
     _inherit = "project.task"
     _populate_sizes = {"small": 500, "medium": 5000, "large": 50000}
-    _populate_dependencies = ["project.project"]
+    _populate_dependencies = ["project.project", "res.partner", "res.users"]
 
     def _populate_factories(self):
         project_ids = self.env.registry.populated_models["project.project"]
         stage_ids = self.env.registry.populated_models["project.task.type"]
+        user_ids = self.env.registry.populated_models['res.users']
+        partner_per_company = {
+            group['company_id'][0]: group['ids']
+            for group in self.env['res.partner']._read_group(
+                domain=[('company_id', '!=', False), ('id', 'in', self.env.registry.populated_models["res.partner"])],
+                fields=['company_id', 'ids:array_agg(id)'],
+                groupby=['company_id'],
+            )
+        }
         def get_project_id(random, **kwargs):
             return random.choice([False, False, False] + project_ids)
         def get_stage_id(random, **kwargs):
             return random.choice([False, False] + stage_ids)
+        def get_partner_id(random, **kwargs):
+            if kwargs['values'].get('project_id'):
+                return random.choice(partner_per_company[kwargs['values']['company_id']])
+            else:
+                return self.env.user.partner_id.id
```

Comment:
```suggestion
        def get_partner_id(random, **kwargs):
            if kwargs['values'].get('project_id'):
                partners = partner_per_company[kwargs['values']['company_id']]
                return random.choice(partners + [False] * len(partners))
            else:
                return False
```
- If the task is private, we can just set the partner to False.
- Let's also have tasks with no partner. :+1:

Reply:
and so `partner_ids`